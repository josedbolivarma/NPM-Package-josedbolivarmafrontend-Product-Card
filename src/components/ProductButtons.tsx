import { useCallback, useContext } from "react";
import { ProductContext } from "./ProductCard";

import styles from '../styles/styles.module.css'
import React from "react";

export interface Props {
    className?: string;
    style?: React.CSSProperties 
}

export const ProductButtons = ({ className, style }: Props) => {
    // TODO: maxCount
    const { increaseBy, counter, maxCount, minCount } = useContext( ProductContext );
    
    // TODO: useCallback isMaxReached = useCallback, dependencias [ counter - cambie, maxCount ]
    const isMaxReached = useCallback(() => !!maxCount && counter === maxCount,
      [counter, maxCount],
    )

    const isMinReached = useCallback(() => !!maxCount && counter === minCount,
    [counter, minCount],
  )

    // TRUE si el count === maxCount
    // FALSE si no lo es
    
    return (
        <div 
            className={ `${ styles.buttonsContainer} ${ className }` }
            style={ style }
        >
            <button
                className={ `${styles.buttonMinus} ${ isMinReached() && styles.disabled }` }
                onClick={ () => increaseBy( -1 ) }> - </button>

            <div className={ styles.countLabel }> { counter } </div>

            <button
                className={ `${styles.buttonAdd} ${ isMaxReached() && styles.disabled }` }
                onClick={ () => increaseBy( +1 ) }> + </button>
        </div>
    );
}