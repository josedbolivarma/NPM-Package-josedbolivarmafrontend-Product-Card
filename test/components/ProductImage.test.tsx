import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard, ProductImage } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage', () => {
    test('debe de mostrar el componente correctamente con la imagen personalizado', () => {

        const wrapper = renderer.create(
            <ProductImage img='https://images.unsplash.com/photo-1505740420928-5e560c06d30e?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8M3x8cHJvZHVjdCUyMGltYWdlfGVufDB8fDB8fA%3D%3D&w=1000&q=80'/>
        )

        expect( wrapper.toJSON() ).toMatchSnapshot();
    });

    test('debe de mostrar el componente con la imagen del producto', () => {
        const wrapper = renderer.create(
            <ProductCard product={ product2 }>
                {
                    () => (
                       <ProductImage /> 
                    )
                }
            </ProductCard>
        )

        expect( wrapper.toJSON() ).toMatchSnapshot();
    })
})