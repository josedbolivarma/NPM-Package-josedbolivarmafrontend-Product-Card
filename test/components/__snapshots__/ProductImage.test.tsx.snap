// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe de mostrar el componente con la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage debe de mostrar el componente correctamente con la imagen personalizado 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="https://images.unsplash.com/photo-1505740420928-5e560c06d30e?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8M3x8cHJvZHVjdCUyMGltYWdlfGVufDB8fDB8fA%3D%3D&w=1000&q=80"
/>
`;